app/frontend/packs/degree-type-autocomplete.js

```diff
@@ -0,0 +1,57 @@
+import accessibleAutocomplete from "accessible-autocomplete";
+
+const initDegreeTypeAutocomplete = () => {
+  function inputTemplate(result) {
+    if (result) {
+      const descriptor = result.split('|');
+
+      return descriptor[1]
+    }
+  }
+
+  function suggestionTemplate(result) {
+    const descriptor = result.split('|');
+
+    if (descriptor[0]) {
+      return `<strong>${descriptor[0]}</strong> <span class="autocomplete__option--hint">${descriptor[1]}</span>`
+    }
+    return `<strong>${descriptor[1]}</strong>`
+  }
+
+  try {
+    const inputIds = [
+      "candidate-interface-degree-type-form-type-description-field",
+      "candidate-interface-degree-type-form-type-description-field-error"
+    ];
+
+    inputIds.forEach(inputId => {
+      const input = document.getElementById(inputId);
+      if (!input) return;
+
+      const containerId = "degree-type-autocomplete";
+      const container = document.getElementById(containerId);
+      if (!container) return;
+
+      const sourceData = JSON.parse(container.dataset.source);
+
+      accessibleAutocomplete({
+        element: container,
+        id: inputId,
+        name: input.name,
+        source: sourceData,
+        showNoOptionsFound: true,
```

Comment:
We should make sure the input is pre-filled with what the user typed in:

```suggestion
        showNoOptionsFound: true,
        defaultValue: input.value,
```